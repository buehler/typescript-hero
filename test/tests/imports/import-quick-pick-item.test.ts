import { ClassDeclaration } from 'typescript-parser';

import ImportQuickPickItem from '../../../src/imports/import-quick-pick-item';
import { expect } from '../setup';

describe('ImportQuickPickItem', () => {

  it('should set the correct fields', () => {
    const item = new ImportQuickPickItem({
      declaration: new ClassDeclaration('class', true),
      from: '/server/indices',
    });

    expect(item).to.matchSnapshot();
  });

  it('should contain the declaration info', () => {
    const item = new ImportQuickPickItem({
      declaration: new ClassDeclaration('class', true),
      from: '/server/indices',
    });

    expect(item.declarationInfo).to.matchSnapshot();
  });

});
