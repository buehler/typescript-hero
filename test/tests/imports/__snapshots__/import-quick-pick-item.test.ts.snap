// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportQuickPickItem should contain the declaration info 1`] = `
Object {
  "declaration": ClassDeclaration {
    "accessors": Array [],
    "end": undefined,
    "isExported": true,
    "methods": Array [],
    "name": "class",
    "properties": Array [],
    "start": undefined,
  },
  "from": "/server/indices",
}
`;

exports[`ImportQuickPickItem should set the correct fields 1`] = `
ImportQuickPickItem {
  "declarationInfo": Object {
    "declaration": ClassDeclaration {
      "accessors": Array [],
      "end": undefined,
      "isExported": true,
      "methods": Array [],
      "name": "class",
      "properties": Array [],
      "start": undefined,
    },
    "from": "/server/indices",
  },
  "description": "/server/indices",
  "label": "class",
}
`;
